app/consumer.py

```diff
@@ -0,0 +1,119 @@
+import base64
+
+from sdx.common.async_consumer import AsyncConsumer
+
+from app import create_and_wrap_logger
+from app import settings
+from app.decrypter import Decrypter, DecryptError
+from app.sdxftp import SDXFTP
+from app.publisher import QueuePublisher
+
+logger = create_and_wrap_logger(__name__)
+
+'''
+The AsyncConsumer will automatically pick up the RABBIT queue from a file called settings with var RABBIT_QUEUE
+TODO this really needs to change
+'''
+QUEUE = settings.RABBIT_QUEUE
+
+
+class Consumer(AsyncConsumer):
+
+    def __init__(self, log=None):
+        super().__init__(log)
+        self._decrypter = Decrypter(settings.RAS_SEFT_PUBLIC_KEY,
+                                    settings.SDX_SEFT_PRIVATE_KEY,
+                                    settings.SDX_SEFT_PRIVATE_KEY_PASSWORD)
+
+        self.quarantine_publisher = QueuePublisher(logger, settings.RABBIT_URLS, settings.RABBIT_QUARANTINE_QUEUE)
+
+        self._ftp = SDXFTP(logger, settings.FTP_HOST, settings.FTP_USER, settings.FTP_PASS, settings.FTP_PORT)
+
+    def on_message(self, unused_channel, basic_deliver, properties, body):
+
+        delivery_count = self.get_delivery_count_from_properties(properties)
+
+        tx_id = self.get_tx_id_from_properties(properties)
+        if not tx_id:
+            self.quarantine_publisher.publish_message(body)
+            self.reject_message(basic_deliver.delivery_tag)
+            logger.error("No tx_id so quarantining message",
+                         action="quarantined",
+                         tx_id=tx_id,
+                         delivery_count=delivery_count)
+
+        encrypted_jwt = body.decode("utf-8")
+        logger.debug("Message Received", tx_id=tx_id)
+
+        try:
+            decrypted_payload = self._decrypter.decrypt(encrypted_jwt)
+        except DecryptError as e:
+            # Move to the quarantine queue
+            self.quarantine_publisher.publish_message(body)
+            self.reject_message(basic_deliver.delivery_tag, tx_id=tx_id)
+            logger.error("Bad decrypt",
+                         action="quarantined",
+                         exception=e,
+                         tx_id=tx_id,
+                         delivery_count=delivery_count)
+        except Exception as e:
+            self.nack_message(basic_deliver.delivery_tag, tx_id=tx_id)
+            logger.error("Failed to process",
+                         action="nack",
+                         exception=e,
+                         tx_id=tx_id,
+                         delivery_count=delivery_count)
+
+        file_contents = decrypted_payload.get("file")
+        file_name = decrypted_payload.get("filename")
+        logger.debug("Decrypted file", file_name=file_name, tx_id=tx_id)
+
+        decoded_contents = base64.b64decode(file_contents)
```

Comment:
Could `file_contents` and `file_name` potentially be `None`? If so, this needs a try/except block.